ext/phar/stream.c

```diff
@@ -44,7 +44,11 @@ php_stream_wrapper_ops phar_stream_wops = {
 	phar_wrapper_unlink,   /* unlink */
 	phar_wrapper_rename,   /* rename */
 	phar_wrapper_mkdir,    /* create directory */
-	phar_wrapper_rmdir,    /* remove directory */
+	phar_wrapper_rmdir     /* remove directory */
+#ifdef PHP_STREAMS_SUPPORT_IS_CACHEABLE
```

Comment:
Thanks for your comments. Now that it is clear that the feature is for PHP7, I’ll remove the define and enclose phar changes in a PHP_VERSION-based ifdef.

 

Regards

 

François

 

De : Stanislav Malyshev [mailto:notifications@github.com] 
Envoyé : dimanche 4 janvier 2015 23:12
À : php/php-src
Cc : Francois Laupretre
Objet : Re: [php-src] Add is_cacheable() stream wrapper operation (#976)

 

In ext/phar/stream.c <https://github.com/php/php-src/pull/976#discussion-diff-22443187> :

> @@ -44,7 +44,11 @@ php_stream_wrapper_ops phar_stream_wops = {
>      phar_wrapper_unlink,   /* unlink */
>      phar_wrapper_rename,   /* rename */
>      phar_wrapper_mkdir,    /* create directory */
> -    phar_wrapper_rmdir,    /* remove directory */
> +    phar_wrapper_rmdir     /* remove directory */
> +#ifdef PHP_STREAMS_SUPPORT_IS_CACHEABLE

I agree, having this define looks pointless - if we accept this RFC, this is part of PHP 7 and should be checked against as "is this PHP 7?" if somebody needs to maintain cross-version extensions. We don't have separate defines for every PHP 7 feature, and it's not configurable in any way, so it should be just there - or, if version portability is needed (not sure for phar, it's a core module, why would it need that?) use PHP version checks. 

—
Reply to this email directly or view it on GitHub <https://github.com/php/php-src/pull/976/files#r22443187> .Image supprimée par l'expéditeur.